graphql_compiler/compiler/emit_sql.py

```diff
@@ -74,30 +77,113 @@ def _find_columns_used(sql_schema_info, ir):
 
     # Find outputs used
     for _, output_info in ir.query_metadata_table.outputs:
+        if isinstance(output_info.location, FoldScopeLocation):
+            continue
         query_path = output_info.location.query_path
         used_columns.setdefault(query_path, set()).add(output_info.location.field)
 
     # Find tags used
     for _, output_info in ir.query_metadata_table.tags:
+        if isinstance(output_info.location, FoldScopeLocation):
+            continue
         query_path = output_info.location.query_path
         used_columns.setdefault(query_path, set()).add(output_info.location.field)
 
     return used_columns
 
 
+def _find_folded_outputs(ir):
+    """For each fold path, find outputs."""
+    folded_outputs = {}
+    # Find outputs used for each fold path
+    for _, output_info in ir.query_metadata_table.outputs:
+        if isinstance(output_info.location, FoldScopeLocation):
+            fold_path = output_info.location.fold_path
+            folded_outputs.setdefault(fold_path, set()).add(output_info.location)
+    return folded_outputs
+
+
+class SQLFoldObject(object):
```

Comment:
Please document what this is and why it exists. Is it a mutable class? I see it being constructed and then mutated by directly going into its private fields. Is that the intended usage?